### PR TITLE
Switch to cert-manager managed certificates by default

### LINKED ISSUE #6: Webhook certificate expires after 30 days of inactivity
By default the component creates the webhook certificate using helm hooks. The helm hook creates the certificate using a CertificateSingingRequest. By default (at least on OpenShift) this certificate is only valid for a month. So if there is no change to the operator for one month the certificate will expire.

## Steps to Reproduce the Problem

  1. Install component
  1. Wait >30 days
  1. Wehook fails

## Actual Behavior

When installing the component with the default settings the operator will stop working after 30 days

## Expected Behavior

When installing the component with the default settings, certificates should either renew themselves or be valid for a very long time.

### PR DESCRIPTION
**Pull request by glrf**
This PR ensures that we actually renew certificates by default by enabling cert-manager support.

The helm chart supports cert-manager certificates by setting `cert.certManager.autoConfigure: true`, which works fine when installing the component on a fresh cluster, however there are some pitfalls when switching to cert-manager certificates:

1. The certificate is generated using a helm hook. This hook doesn't run unless ArgoCD sees a change in the actually deployed resources. That's why I added the cert configuration to the operator deployment as an annotation
2. There is a race-condition in the setup jobs. If the certificate secret already exists, the job will write the old certificate in the validation webhook config, which will lead to a certificate error until the sync run a second time. I documented the manual steps. Only removing the secrets first worked reliably.

Fixes #6 
## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

**Review by Kidswiss — APPROVED** [review]
Generally LGTM

Just one thought: the migration from existing to managed certificates is a one time event, right? Given we currently have two instances deployed, and any future instance will have this enabled by default. Does it really warrant to have ugly reclass hacks in the component vs. documenting how to fix it manually?

**Comment by glrf** [comment]
> Just one thought: the migration from existing to managed certificates is a one time event, right? Given we currently have two instances deployed, and any future instance will have this enabled by default. Does it really warrant to have ugly reclass hacks in the component vs. documenting how to fix it manually?

Yes the issue is only when switching. Maybe you're right and I over-engineered this. I don't like when components need extra manual steps to modify some config, but I guess changing the way we generate certificates happen rarely enough that it doesn't justify the added complexity.

I think you're right. I'll remove it

**Review by zugao — APPROVED** [review]
Just a doc issue other than that LGTM

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 1]
```diff
@@ -14,12 +14,24 @@ The namespace in which to deploy this component.
...
+
+[WARNING]
+====
+This means by default this component requires a working cert-manager instance on the cluster.
+If cert-manager isn't installed on the target cluster, disable cert-manager support by setting `helmValues.cert.certManager.autoConfigure: false`.
+====
+
+[IMPORTANT]
+====
+If you need to switch from or the cert-manager generated certificates you will have to manually remove the secrets `stackgres-operator-certs` and `stackgres-operator-ca` *before* applying the catalog.
```
Comment (zugao) [inline docs/modules/ROOT/pages/references/parameters.adoc]: This phrase seems broken, I cannot understand it. Can you rephrase it a bit better?

**Comment by Kidswiss** [comment]
> > Just one thought: the migration from existing to managed certificates is a one time event, right? Given we currently have two instances deployed, and any future instance will have this enabled by default. Does it really warrant to have ugly reclass hacks in the component vs. documenting how to fix it manually?
> 
> Yes the issue is only when switching. Maybe you're right and I over-engineered this. I don't like when components need extra manual steps to modify some config, but I guess changing the way we generate certificates happen rarely enough that it doesn't justify the added complexity.
> 
> I think you're right. I'll remove it

I generally agree with that sentiment, but it's always about finding the right balance.

File: docs/modules/ROOT/pages/references/parameters.adoc (hunk 1 above)
Comment (glrf) [reply docs/modules/ROOT/pages/references/parameters.adoc]: Is this clearer?
